### PR TITLE
Add : Docker Support

### PR DESCRIPTION
**Pull request by nandithags11**
## Description

Adding prod - leve docker to the repo for backend for now

Add Docker, multi stage build, CI pipeline to publish with workflow_dispatch only
Required for release and stuff
N/A
---

## Type of Change

- [ ] Bug fix
- [ X ] New feature
- [ ] Refactor / code cleanup
- [ ] Performance improvement
- [ X ] CI / tooling change
- [ ] Documentation update
- [ ] Security fix

---

## Related Issues

Closes # N/A

---

## Backend Checklist

- [ ] Models: migrations created and reviewed (`makemigrations` + `migrate`)
- [ ] Serializers: input validation is correct; no sensitive fields exposed
- [ ] Views: correct permission classes applied (`IsAdmin`, `IsShopOwner`, `IsShopOwnerOrAdmin`)
- [ ] URLs: new endpoints registered in `urls.py`
- [ ] All responses use `APIResponse` wrapper
- [ ] No raw SQL or unfiltered querysets exposed to unauthenticated users
- [ X ] `manage.py check` passes locally

---

## Tests

- [ ] Wrote tests for new functionality N/A
- [ X ] Existing tests still pass 
- [ ] Tested happy path and edge cases manually

Describe what you tested and how:

```
Ran dockerbuild in Mac OS
```

---

## Docs / Notes

N/A

### CODE REVIEW

File: toddy_shop_backend/config/settings.py [hunk 1]
```diff
@@ -113,6 +116,16 @@
...
 
 STATIC_URL = "static/"
+STATIC_ROOT = BASE_DIR / "staticfiles"
+
+STORAGES = {
+    "default": {
+        "BACKEND": "django.core.files.storage.FileSystemStorage",
+    },
+    "staticfiles": {
+        "BACKEND": "whitenoise.storage.CompressedManifestStaticFilesStorage",
```
Comment (abhi-io) [inline toddy_shop_backend/config/settings.py:126]: hi, is the 'whitenoice' needed? is it our architecture?
